src/main/java/uk/gov/ch/repository/officers/ActiveOfficerDetailsRepository.java

```diff
@@ -0,0 +1,58 @@
+package uk.gov.ch.repository.officers;
+
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.core.PreparedStatementSetter;
+import org.springframework.stereotype.Repository;
+
+import uk.gov.ch.OracleQueryApplication;
+import uk.gov.ch.exception.NoActiveOfficersFoundException;
+import uk.gov.ch.model.officer.active.ActiveOfficerDetails;
+import uk.gov.companieshouse.logging.Logger;
+import uk.gov.companieshouse.logging.LoggerFactory;
+
+@Repository
+public class ActiveOfficerDetailsRepository {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(OracleQueryApplication.APPLICATION_NAME_SPACE);
+
+    @Autowired
+    private JdbcTemplate jdbcTemplate;
+
+    static final String OFFICER_DETAILS_SQL = "SELECT "
```

Comment:
It looks like we could return a list of officers but we will end up throwing an exception from the service if there are more than 1, which will be inefficient as some companies could have a large amount of officers. there should be a way in SQL to only return the list if the officer count is 1, this way oracle will do the hard work for us and we don't have to transfer all the unneeded data. 
My SQL is rusty but I'd recommend consulting with a DBA (Mike Williams?) as to how to do that. It might be possible to add a 'where COUNT(officer something) = 1' or something along those lines.

Reply:
i was thinking something like this could be used - https://stackoverflow.com/a/21709485 where we could join with the selection of single officers

Reply:
probably best to ask the DBA to recommend a way to only return for companies with 1 officer, i'm no sql expert!

Reply:
been having a go and I seem to have it only returning data if the company only has 1 officer.

add this to the where clause - 
`AND cba.corporate_body_id in (  SELECT corporate_body_id
         FROM corporate_body_appointment
         GROUP BY corporate_body_id
         HAVING COUNT(corporate_body_id) = 1)`

Performance seems similar to the query before but i've been running it several times so it will cache the query. The first time it runs might be slow so worth checking. Also get a DBA to verify it , maybe they can optimise the query?